### DISCUSSION TITLE
Struggling making changes to generated image

### DISCUSSION THREAD

**Discussion by kpagcha** [lllyasviel/Fooocus#1153] [Q&A]
I am trying to wrap my head around how tweaking images work. I have an image of Bulbasaur-inspired creature but it has a weird black dot on its face which I am trying to remove. So, as you can see in the screenshot, I selected it with the inpaint tool, chose "modify content" and in the prompt indicated "remove black dot".

![imagen](https://github.com/lllyasviel/Fooocus/assets/3865095/9173400e-c643-4aea-a725-e64a0eaacf9e)

However, the most random image comes out:

![imagen](https://github.com/lllyasviel/Fooocus/assets/3865095/6a06c0a7-098a-44db-99f7-02d20d5473e7)

What am I doing wrong?

Bonus question: I am also stuggling with image prompts. For example, what if I want the creature to have longer legs? Or a different skin tone? Am I supposed to use the inpaint tool (which so far I can't get to work to make modifications)? Or do I just say "longer legs"?

**Comment by ut1kaa** [comment]
It would be better to write something like "blue skin" because he's not likely to understand your request.

**Comment by ut1kaa** [comment]
To change an object, paint over the area and enter what you want to see in the prompt. For example, paint over the skin and write green skin, or paint over the legs and the area next to them and write long legs, it should work.

**Reply by kpagcha** [reply]
This is the result if I paint over the dot and prompt "dark green screen" in the additional prompt lol.

![image](https://github.com/lllyasviel/Fooocus/assets/3865095/2b0f5084-f7c0-4b92-a801-30d5258bee1b)

![imagen](https://github.com/lllyasviel/Fooocus/assets/3865095/4b0188be-550b-451a-a052-d103b646f73f)

**Comment by mel870** [comment]
Depending on the changes you're trying to make, it's sometimes quicker to edit in Photoshop, Photopea etc. Especially if you just need to remove versus add. I personally use Photoshop generative fill for certain edits. However, if you need to inpaint to edit the image, and you're getting completely diffferent results, enable advanced settings. Then enable the option 'Developer Debug Mode' and check 'Mixing Image Prompt and Inpaint' under Control. Then set the image you created in the Image Prompt section and enable Canny  before inpainting. Then adjust the canny weight and stop until you get desired results.

**Comment by valcar3d** [comment]
If you want slight changes and maintain the structure you should try to use the advanced settings of image prompt and set PyraCanny option
![Untitled](https://github.com/lllyasviel/Fooocus/assets/15023176/05ab6ed4-10a8-4443-8b33-900995d3c161)

**Reply by kpagcha** [reply]
How can I use that to remove the dot? I understand PyraCanny/CPDS basically keeps the structure and is used for things like ambience changes. But specific changes like what I am after?
